src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py

```diff
@@ -90,6 +146,22 @@ def on_rpc_complete(self, request_id: int, peer: str, method: str) -> None:
                 else:
                     self._rpcs_by_peer[peer] += 1
                     self._rpcs_by_method[method][peer] += 1
+                    rpc_metadata = (
```

Comment:
Wrap it into `if self._metadata_keys:` to avoid building all this in tests with no metadata configured